### PR TITLE
DO_NOT_MERGE cmd/gitannex: Add e2e test to attempt to repro Dropbox i…

### PR DESCRIPTION
**Pull request by dmcardle**
…ssue

Context: https://git-annex.branchable.com/bugs/Trust_but_Verify__58___RClone/

The e2e test in this commit should not be merged! I've hardcoded a few values specific to my own system, specifically the rclone remote name and the path to my rclone config.

#### Checklist

- [ ] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [ ] I have added tests for all changes in this PR if appropriate.
- [ ] I have added documentation for the changes if appropriate.
- [ ] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [ ] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Comment by dmcardle** [comment]
Oh yeah, `fstest`  should definitely be on the agenda! I'll file an issue.

This was just my quick attempt to repro the git-annex bug report linked above. Still unclear where the bad behavior comes from: presumably it's either rclone, git-annex proper, the Dropbox client, or some interaction between them. It will definitely be valuable to generalize these tests to use fstest.

**Comment by Servinjesus1** [comment]
@dmcardle Spencer here - from that bug report. I think it's safe to assume my issue was a one-off oddity, maybe due to my use of Dropbox's finnicky sync utility. In another test drop successfully checked RClone and refused to drop after seeing the file had been moved. However, no other git annex command seems to notice the file has been dropped. `git annex findkeys --in rcloneremote` nor `git annex whereis` seemed to indicate the file was gone even after I ran `git annex sync rcloneremote`. In other words:

1. In one repo I moved files to my rclone remote.
2. In another repo, I added the rclone remote and moved the files out of it
3. Back in the first repo, I ran `git annex sync rcloneremote`
4. Both `git annex whereis` and `git annex findkeys --in rcloneremote` continued to show the files as in the rclone.

So basically, it seems like rclone doesn't inform git-annex of the existence of files on rclone remotes for normal git-annex purposes and only provides a drop check. If this is true I might want to add a feature request to rclone gitannex for the capability of updating key information from rclone remotes. We can definitely break out discussion on this elsewhere if you prefer. Appreciate your help so far!

**Comment by ncw** [comment]
@dmcardle I'll close this since I think it has outlived its purpose :-)
